### PR TITLE
kubectl: Move generators into a subpackage

### LINKED ISSUE #36805: Split generator files into its own subpackage
`pkg/kubectl` is getting messier and messier with the number of generators being created. I'm propose creating new subpackage `generators` and split each generator into its own file.

@fabianofranz @kubernetes/kubectl opinions?

### PR DESCRIPTION
**Pull request by tcharding**
**What this PR does / why we need it**:

There are a bunch of generators within `kubectl`. Code organization would improve if we had a sub-package for generators.

This PR creates a sub-package `generators` and moves generator code from `kubectl` into the newly created sub-package. Updates call sites within `kubectl/cmd`. 

Some other code had to be moved around also to maintain cycle free imports.

**Which issue this PR fixes** 

fixes #36805

**Special notes for your reviewer**:

This is a big PR, I could think of no clear way to split it up any smaller. Suggestions welcome. 

**Release note**:

```release-note
NONE
```
/sig cli
/kind cleanup

### CODE REVIEW

File: pkg/kubectl/BUILD [hunk 1]
```diff
@@ -9,55 +9,40 @@ load(
...
         "//pkg/apis/extensions:go_default_library",
-        "//pkg/apis/rbac:go_default_library",
         "//pkg/client/clientset_generated/internalclientset:go_default_library",
         "//pkg/client/clientset_generated/internalclientset/fake:go_default_library",
         "//pkg/client/clientset_generated/internalclientset/typed/batch/internalversion:go_default_library",
         "//pkg/client/clientset_generated/internalclientset/typed/core/internalversion:go_default_library",
         "//pkg/client/clientset_generated/internalclientset/typed/extensions/internalversion:go_default_library",
         "//pkg/kubectl/util:go_default_library",
         "//pkg/printers:go_default_library",
+<<<<<<< HEAD
```
Comment (xiangpengzhao) [inline pkg/kubectl/BUILD]: Please fix the conflict when rebasing.

File: pkg/kubectl/BUILD [hunk 2]
```diff
@@ -131,26 +93,25 @@ go_library(
...
-        "//pkg/kubectl/util/hash:go_default_library",
         "//pkg/kubectl/util/slice:go_default_library",
         "//pkg/printers:go_default_library",
         "//pkg/printers/internalversion:go_default_library",
         "//vendor/github.com/emicklei/go-restful-swagger12:go_default_library",
         "//vendor/github.com/golang/glog:go_default_library",
         "//vendor/github.com/spf13/cobra:go_default_library",
-        "//vendor/github.com/spf13/pflag:go_default_library",
         "//vendor/k8s.io/api/apps/v1beta1:go_default_library",
+<<<<<<< HEAD
```
Comment (xiangpengzhao) [inline pkg/kubectl/BUILD]: and here.

**Review by xiangpengzhao — CHANGES_REQUESTED** [review]
please be sure to fix all the conflicts when rebasing.

File: pkg/kubectl/BUILD (hunk 2 above)
Comment (tcharding) [reply pkg/kubectl/BUILD]: My bad. Thanks for pointing it out.

**Comment by tcharding** [comment]
lol, 14 `lint`ing pull requests last week and I forgot to run lint on this one.

**Comment by tcharding** [comment]
I believe the visibility declaration needs updating. I am unsure how to go about this.
```
ERROR: /home/tobin/build/go/src/k8s.io/kubernetes/pkg/kubectl/generators/BUILD:3:1: Target '//pkg/kubectl/util:go_default_library' is not visible from target '//pkg/kubectl/generators:go_default_library'. Check the visibility declaration of the former target if you think the dependency is legitimate
```

**Comment by tcharding** [comment]
Why `pull-kubernetes-verify` is failing on `hack/verfiy-golint.sh` has me stumped. `golint` does not throw any errors when run on `pkg/kubernetes/generators/*.go` locally.

**Comment by xiangpengzhao** [comment]
@tcharding you can check the details:
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/51630/pull-kubernetes-verify/50227/

```
I0906 06:03:32.287] Verifying hack/make-rules/../../hack/verify-golint.sh
W0906 06:04:38.400] Errors from golint:
W0906 06:04:38.401] pkg/kubectl/constants/constants.go:21:2: exported const ServiceAccountNameAnnotation should have comment (or a comment on this block) or be unexported
W0906 06:04:38.401] 
W0906 06:04:38.402] Please review the above warnings. You can test via "golint" and commit the result.
W0906 06:04:38.402] If the above warnings do not make sense, you can exempt this package from golint
W0906 06:04:38.402] checking by adding it to hack/.golint_failures (if your reviewer is okay with it).
W0906 06:04:38.402] 
I0906 06:04:38.503] FAILED   hack/make-rules/../../hack/verify-golint.sh	66s
```

**Pull request closed without merging**